### PR TITLE
MM-21976: Include cache layer to be tested

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

This PR includes multiple parts:

- It adds the cache layer on top of the storage to also test code that uses the cache layer
- Fixes test or code that is failing because of that

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21976

### CODE REVIEW

File: api4/user_test.go [hunk 1]
```diff
@@ -2281,34 +2280,43 @@ func TestUpdateUserPassword(t *testing.T) {
 
 	password := "newpassword1"
 	pass, resp := th.Client.UpdateUserPassword(th.BasicUser.Id, th.BasicUser.Password, password)
+	th.App.InvalidateCacheForUser(th.BasicUser.Id)
```
Comment (agnivade) [inline api4/user_test.go]: Shouldn't this invalidation happen in the server side app layer ? I see `InvalidateCacheForUser` being called in several places in app/user.go. Perhaps it is missing this call in the `UpdatePassword` method ?

I mean what happens if an actual client calls this method ? They should not need to manually invalidate the cache right ?

Comment (ethervoid) [reply api4/user_test.go]: Hum, you're right. I've misread that part thinking it was just an internal method. I'm going to check. Good point. Thank you

**Comment by ethervoid** [comment]
@agnivade Yup, added @jespino as reviewer

**Pull request merged**